src/schnetpack/interfaces/ase_interface.py

```diff
@@ -341,6 +341,289 @@ def calculate(
+        self.converter = converter(
+            neighbor_list=neighbor_list,
+            device=device,
+            dtype=dtype,
+            transforms=transforms,
+        )
+
+        self.energy_key = energy_key
+        self.force_key = force_key
+        self.stress_key = stress_key
+
+        # Mapping between ASE names and model outputs
+        self.property_map = {
+            self.energy: energy_key,
+            self.forces: force_key,
+            self.stress: stress_key,
+        }
+
+        # set up basic conversion factors
+        self.energy_conversion = convert_units(energy_unit, "eV")
+        self.position_conversion = convert_units(position_unit, "Angstrom")
+
+        # Unit conversion to default ASE units
+        self.property_units = {
+            self.energy: self.energy_conversion,
+            self.forces: self.energy_conversion / self.position_conversion,
+            self.stress: self.energy_conversion / self.position_conversion**3,
+        }
+
+        # Load multiple models
+        self.models = [self._load_model(model_file) for model_file in models]
+
+        if uncertainty_fn is None:
+            self.uncertainty_fn = self.AbsoluteUncertainty
+        else:
+            self.uncertainty_fn = uncertainty_fn.__get__(self, type(self))
+
+    def calculate(
+        self,
+        atoms: ase.Atoms = None,
+        properties: List[str] = ["energy"],
+        system_changes: List[str] = all_changes,
+        energy_weight: float = 0,
+        force_weight: float = 1.0,
+        stress_weight: float = 0,
+    ):
+        """
+        Calculate properties by averaging results from multiple models.
+        """
+        Calculator.calculate(self, atoms)
+
+        model_inputs = self.converter(atoms)
+        accumulated_results = {prop: [] for prop in properties}
+
+        for model in self.models:
+            model_results = model(model_inputs)
+
+            for prop in properties:
+                model_prop = self.property_map[prop]
+                if model_prop in model_results:
+                    value = model_results[model_prop].cpu().data.numpy()
+                    if prop == self.energy:
+                        value = value.item()
+                    elif prop == self.stress:
+                        value = value.squeeze()
+                    accumulated_results[prop].append(value * self.property_units[prop])
+                else:
+                    raise AtomsConverterError(
+                        f"'{prop}' is not a property of your models. Please check the model properties!"
+                    )
+
+        # Compute average values
+        self.results = {
+            prop: np.mean(accumulated_results[prop], axis=0) for prop in properties
+        }
+        # Compute uncertainty using assigned uncertainty function
+        self.results["uncertainty"] = self.uncertainty_fn(
+            predictions=accumulated_results,
+            energy_weight=energy_weight,
```

Comment:
When `uncertainty_fn` is transformed to a class, the `xxx_weight` can be removed here

Reply:
see comment in line 500